src/java/org/apache/cassandra/concurrent/Stage.java

```diff
@@ -124,10 +128,20 @@ public static Stage fromPoolName(String stageName)
     // Convenience functions to execute on this stage
     public void execute(Runnable command) { executor().execute(command); }
     public void execute(Runnable command, ExecutorLocals locals) { executor().execute(command, locals); }
-    public void maybeExecuteImmediately(Runnable command) { executor().maybeExecuteImmediately(command); };
+    public void maybeExecuteImmediately(Runnable command) { executor().maybeExecuteImmediately(command); }
     public <T> Future<T> submit(Callable<T> task) { return executor().submit(task); }
     public Future<?> submit(Runnable task) { return executor().submit(task); }
     public <T> Future<T> submit(Runnable task, T result) { return executor().submit(task, result); }
+    public static Future<?> submitToMigrationIfNotShutdown(Runnable task)
+    {
+        if (MIGRATION.executor().isShutdown() || MIGRATION.executor().isTerminated())
+        {
+            logger.info("Skipped scheduled pulling schema from other nodes: the MIGRATION executor service has been shutdown.");
```

Comment:
At this point, I think we have a correct solution, but again, there are a couple things that happen on the `MIGRATION` stage that aren't strictly schema pulls, so it may be a little cleaner for now to keep this method in `MigrationCoordinator`.

Reply:
I will address this in the final commit after I squash everything, thank you. I agree with what you said.